platform-validation/scripts/compatibility.sh

```diff
@@ -0,0 +1,418 @@
+            fi
+            ;;
+
+        ppc)
+            _cputype=powerpc
+            ;;
+
+        ppc64)
+            _cputype=powerpc64
+            ;;
+
+        ppc64le)
+            _cputype=powerpc64le
+            ;;
+
+        s390x)
+            _cputype=s390x
+            ;;
+
+        *)
+            err "unknown CPU type: $_cputype"
+
+    esac
+
+    # Detect 64-bit linux with 32-bit userland
+    if [ "${_ostype}" = unknown-linux-gnu ] && [ "${_bitness}" -eq 32 ]; then
+        case $_cputype in
+            x86_64)
+                _cputype=i686
+                ;;
+            mips64)
+                _cputype=$(get_endianness mips '' el)
+                ;;
+            powerpc64)
+                _cputype=powerpc
+                ;;
+            aarch64)
+                _cputype=armv7
+                if [ "$_ostype" = "linux-android" ]; then
+                    _ostype=linux-androideabi
+                else
+                    _ostype="${_ostype}eabihf"
+                fi
+                ;;
+        esac
+    fi
+
+    # Detect armv7 but without the CPU features Rust needs in that build,
+    # and fall back to arm.
+    # See https://github.com/rust-lang/rustup.rs/issues/587.
+    if [ "$_ostype" = "unknown-linux-gnueabihf" ] && [ "$_cputype" = armv7 ]; then
+        if ensure grep '^Features' /proc/cpuinfo | grep -q -v neon; then
+            # At least one processor does not have NEON.
+            _cputype=arm
+        fi
+    fi
+
+    OSTYPE=$_ostype
+    ARCH=$_cputype
+}
+
+perform_cleanup(){
+    rm -rf cap.txt || true
+    #TODO : Cleanup docker images
+}
+
+
+# ------------------------------------------------------------------------------
+# Check whether the Target Device can be used to set capability. EdgeHub Runtime component sets CAP_NET_BIND which is Required
+# for Azure IoT Edge Operation
+# ------------------------------------------------------------------------------
+perform_capability_check_host(){
+    
+    wrap_debug "Checking Set Cap Capability on Host.."
+    touch cap.txt
+    setcap "cap_net_bind_service=+ep" cap.txt
+    if [ $? != 0 ]; then
+        #TODO Check Mark Failed in Red
+        wrap_debug "Failed to Set Capability on Host Container"
+        wrap_fail "Capability_Check_Host" "Fail"
+        return
+    fi
+
+    contains=$(getcap cap.txt | grep 'cap_net_bind_service+ep')
+    if [ $? != 0 ] && [ -z "${contains##*cap_net_bind_service+ep*}" ]; then
+        wrap_debug "Failed to Verify Set Capability on Host Container"
+        wrap_fail "Capability_Check_Host" "Fail"
+        return
+    fi
+
+    wrap_pass "Capability_Check_Host" "Pass"
+}
+
+perform_capability_check_container(){
+    #Check For Docker
+    wrap_debug "Checking Set Cap Capability on Container.."
+    need_cmd docker
+    CAP_CMD="getcap cap.txt"
+    DOCKER_VOLUME_MOUNTS=''
+    DOCKER_IMAGE="ubuntu:18.04"
+    #TODO: Use different images based on platform and arch
```

Comment:
What different images would you run?

Actually, [api-proxy on master](https://github.com/Azure/iotedge/blob/master/edge-modules/api-proxy-module/docker/linux/amd64/Dockerfile#L6-L10) might be a good example for how to set up an Alpine container:

```
FROM alpine:3.14.0
RUN apk update && apk add --no-cache libcap
```

Reply:
I realized that I do not need to run different images, simply doing a docker pull on ubuntu:18.04 works for amd64,arm64 and armv7. 
Do you think we need to worry about other architectures?

Reply:
I don't think so, at least not to start. Using alpine instead of ubuntu will make the script faster (because the alpine image is smaller and will download faster), but we can do that later, as an optimization.